docs/GPT_session_arch.md

```
@@ -0,0 +1,291 @@
+
+export function parseToolCallResponse(content: string): ParsedResponse {
+  if (!content?.trim()) {
+    return { isToolCall: false, toolCalls: [], textResponse: "" };
+  }
+
+  // 1. Match JSON in markdown fences
+  const jsonBlockMatch = content.match(/```json\s*([\s\S]*?)```/);
+  let candidate = jsonBlockMatch ? jsonBlockMatch[1] : content.trim();
+
+  // 2. Try direct JSON parsing
+  try {
```

markdownlint-cli2 (0.18.1)

Spaces inside emphasis markers

(MD037, no-space-in-emphasis)